pysph/tools/binder.py

```diff
@@ -0,0 +1,208 @@
+    # It is a valid regex sequence that matches to a digit.
+
+    match_list = []
+    with open(glob.glob(log_file_path)[0], 'r') as file:
+        for line in file:
+            for match in re.finditer(regex, line, re.S):
+                match_text = match.group()
+                match_list.append(match_text)
+                if len(match_list) > 0:
+                    break
+            if len(match_list) > 0:
+                break
+    return match_list[0][-1] + 'D'
+
+
+def make_notebook(path, sim_name, config_dict={}):
+    '''
+    Makes a jupyter notebook to view simulation results stored in
+    a given directory
+
+    path: the directory conatining the output files
+
+    sim_name: name of the simulation
+              ex. 'cavity_output'
+
+    config_dict: configuration dictionary for the notbeook viewer [dict]
+                 ex. {'fluid': {'frame': 20}}
+    '''
+
+    viewer_type = find_viewer_type(path)
+    cell1_src = [
+        "import os\n",
+        "from pysph.tools.ipy_viewer import Viewer" + viewer_type
+    ]
+    cell2_src = [
+        "cwd = os.getcwd()"
+    ]
+    cell3_src = [
+        "viewer = Viewer" + viewer_type + "(cwd)\n",
+        "viewer.interactive_plot(" + str(config_dict) + ")"
+    ]
+
+    nb = nbf.v4.new_notebook()
+
+    nb.cells = [
+        nbf.v4.new_code_cell(source=cell1_src),
+        nbf.v4.new_code_cell(source=cell2_src),
+        nbf.v4.new_code_cell(source=cell3_src)
+    ]
+
+    nbf.write(nb, path+'/'+sim_name+'.ipynb')
+    return
+
+
+def find_sim_dirs(path, sim_paths_list=[]):
+    '''
+    Finds all the directories in a given directory that
+    contain pysph output files.
+    '''
+    path = os.path.abspath(path)
+    sim_files = get_files(path)
+    if len(sim_files) != 0:
+        sim_paths_list.append(path)
+    elif len(sim_files) == 0:
+        files = os.listdir(path)
+        files = [f for f in files if os.path.isdir(f)]
+        files = [os.path.abspath(f) for f in files if not f.startswith('.')]
+        for f in files:
+            sim_paths_list = find_sim_dirs(f, sim_paths_list)
+
+    return sim_paths_list
+
+
+def find_dir_size(path):
+    '''
+    Finds the size of a given directory.
+    '''
+    total_size = 0
+    for dir_path, dir_names, file_names in os.walk(path):
+        for f in file_names:
+            fp = os.path.join(dir_path, f)
+            if not os.path.islink(fp):
+                total_size += os.path.getsize(fp)
+
+    return total_size
+
+
+def make_binder(path):
+
+    src_path = os.path.abspath(path)
+
+    sim_paths_list = find_sim_dirs(src_path)
+
+    for path in sim_paths_list:
+        sim_name = os.path.split(path)[1]
+        make_notebook(path, sim_name)
+        if len(sim_paths_list) == 1 and sim_paths_list[0] == src_path:
+            files = os.listdir(src_path)
+            files = [src_path+'/'+f for f in files]
+            mkdir(src_path+'/'+sim_name)
```

Comment:
Use `os.path.join` here and other places also.

Reply:
Fixed.